runtime/expr/function/regexp.go

```diff
@@ -41,3 +41,35 @@ func (r *Regexp) Call(ctx zed.Allocator, args []zed.Value) *zed.Value {
 	}
 	return ctx.NewValue(r.typ, r.builder.Bytes())
 }
+
+// https://github.com/brimdata/zed/blob/main/docs/language/functions.md#regexp_replace
+type RegexpReplace struct {
+	zctx  *zed.Context
+	re    *regexp.Regexp
+	restr string
+	err   error
+}
+
+func (r *RegexpReplace) Call(ctx zed.Allocator, args []zed.Value) *zed.Value {
+	sVal := args[0]
+	reVal := args[1]
+	newVal := args[2]
+	if !sVal.IsString() || !reVal.IsString() || !newVal.IsString() {
+		return newErrorf(r.zctx, ctx, "regexp_replace: string arg required")
+	}
+	if sVal.Bytes == nil {
+		return zed.Null
+	}
+	if reVal.Bytes == nil || newVal.Bytes == nil {
+		return newErrorf(r.zctx, ctx, "regexp_replace: an input arg is null")
+	}
+	re := zed.DecodeString(reVal.Bytes)
+	if r.restr != re {
+		r.restr = re
+		r.re, r.err = regexp.Compile(re)
+	}
+	if r.err != nil {
+		return newErrorf(r.zctx, ctx, "regexp_replace: %s", r.err)
+	}
+	return ctx.NewValue(zed.TypeString, r.re.ReplaceAllLiteral(sVal.Bytes, newVal.Bytes))
```

Comment:
🤷🏻‍♂️ I wasn't sure if this was something someone would normally expect to work in for replace in regular expressions or if this was a go specific feature.

Reply:
Also seems like if you were unaware of this and your replace had a $ in it it might cause odd behavior, felt like it was safer to not use this and wait for people to ask for this.